packages/editor-ui/src/views/NodeView.vue

```diff
@@ -3957,6 +3957,28 @@ export default defineComponent({
 				tempWorkflow.renameNode(oldName, nodeNameTable[oldName]);
 			}
 
+			if (data.pinData) {
+				let pinDataSuccess = true;
+				for (const nodeName of Object.keys(data.pinData)) {
+					// Pin data limit reached
+					if (!pinDataSuccess) {
+						this.showError(
+							new Error(this.$locale.baseText('ndv.pinData.error.tooLarge.description')),
+							this.$locale.baseText('ndv.pinData.error.tooLarge.title'),
+						);
+						continue;
```

Comment:
1. Shouldn't this be `break`? Once we reach the limit, we know this will fail for all the following nodes.
2. Idea: We could sort pin data by size in ascending order, to ensure we set the most nodes we can with pin data.